<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>942</width>
    <height>640</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Modbus仿真器</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>932</width>
      <height>631</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="minimumSize">
        <size>
         <width>580</width>
         <height>180</height>
        </size>
       </property>
       <property name="title">
        <string>设置</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>连接</string>
          </property>
          <widget class="QTabWidget" name="tabConnect">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>381</width>
             <height>101</height>
            </rect>
           </property>
           <property name="tabPosition">
            <enum>QTabWidget::TabPosition::North</enum>
           </property>
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="tabTcp">
            <attribute name="title">
             <string>TCP</string>
            </attribute>
            <widget class="QLabel" name="label_3">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>61</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>从站IP</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_4">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>50</y>
               <width>53</width>
               <height>15</height>
              </rect>
             </property>
             <property name="text">
              <string>端口</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="txtIp">
             <property name="geometry">
              <rect>
               <x>70</x>
               <y>10</y>
               <width>201</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>127.0.0.1</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="txtPort">
             <property name="geometry">
              <rect>
               <x>70</x>
               <y>40</y>
               <width>201</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>502</string>
             </property>
            </widget>
            <widget class="QPushButton" name="btnTcp">
             <property name="geometry">
              <rect>
               <x>290</x>
               <y>10</y>
               <width>80</width>
               <height>51</height>
              </rect>
             </property>
             <property name="text">
              <string>连接</string>
             </property>
            </widget>
           </widget>
           <widget class="QWidget" name="tabRtu">
            <attribute name="title">
             <string>RTU</string>
            </attribute>
            <widget class="QComboBox" name="cbxCom">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>81</width>
               <height>23</height>
              </rect>
             </property>
             <item>
              <property name="text">
               <string>刷新</string>
              </property>
             </item>
            </widget>
            <widget class="QLineEdit" name="txtBaud">
             <property name="geometry">
              <rect>
               <x>100</x>
               <y>10</y>
               <width>151</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>115200</string>
             </property>
            </widget>
            <widget class="QComboBox" name="cbxData">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>40</y>
               <width>81</width>
               <height>23</height>
              </rect>
             </property>
             <item>
              <property name="text">
               <string>8</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>7</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>6</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>5</string>
              </property>
             </item>
            </widget>
            <widget class="QComboBox" name="cbxParity">
             <property name="geometry">
              <rect>
               <x>100</x>
               <y>40</y>
               <width>72</width>
               <height>23</height>
              </rect>
             </property>
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Even</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Odd</string>
              </property>
             </item>
            </widget>
            <widget class="QComboBox" name="cbxStop">
             <property name="geometry">
              <rect>
               <x>180</x>
               <y>40</y>
               <width>72</width>
               <height>23</height>
              </rect>
             </property>
             <item>
              <property name="text">
               <string>1</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2</string>
              </property>
             </item>
            </widget>
            <widget class="QPushButton" name="btnOpenCom">
             <property name="geometry">
              <rect>
               <x>270</x>
               <y>10</y>
               <width>101</width>
               <height>51</height>
              </rect>
             </property>
             <property name="text">
              <string>打开串口</string>
             </property>
            </widget>
           </widget>
          </widget>
          <widget class="QLabel" name="label_2">
           <property name="geometry">
            <rect>
             <x>190</x>
             <y>20</y>
             <width>41</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>模式</string>
           </property>
          </widget>
          <widget class="QRadioButton" name="rbtnModeMaster">
           <property name="geometry">
            <rect>
             <x>240</x>
             <y>20</y>
             <width>70</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>主站</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QRadioButton" name="rbtnModeSlave">
           <property name="geometry">
            <rect>
             <x>320</x>
             <y>20</y>
             <width>70</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>从站</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>modbus属性</string>
          </property>
          <widget class="QLabel" name="label">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>53</width>
             <height>25</height>
            </rect>
           </property>
           <property name="text">
            <string>从站ID</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="txtSlaveId">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>30</y>
             <width>71</width>
             <height>25</height>
            </rect>
           </property>
           <property name="text">
            <string>1</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_5">
           <property name="geometry">
            <rect>
             <x>160</x>
             <y>30</y>
             <width>53</width>
             <height>25</height>
            </rect>
           </property>
           <property name="text">
            <string>功能码</string>
           </property>
          </widget>
          <widget class="QComboBox" name="cbxFuncode">
           <property name="geometry">
            <rect>
             <x>220</x>
             <y>30</y>
             <width>141</width>
             <height>25</height>
            </rect>
           </property>
           <item>
            <property name="text">
             <string>3(hold register)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4(input register)</string>
            </property>
           </item>
          </widget>
          <widget class="QLabel" name="label_6">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>70</y>
             <width>41</width>
             <height>25</height>
            </rect>
           </property>
           <property name="text">
            <string>地址</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_7">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>100</y>
             <width>41</width>
             <height>25</height>
            </rect>
           </property>
           <property name="text">
            <string>数量</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="txtAddr">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>70</y>
             <width>113</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>0</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="txtRegisterCnt">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>100</y>
             <width>113</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>10</string>
           </property>
          </widget>
          <widget class="QPushButton" name="btnConfirmConfig">
           <property name="geometry">
            <rect>
             <x>280</x>
             <y>70</y>
             <width>81</width>
             <height>51</height>
            </rect>
           </property>
           <property name="text">
            <string>确认</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QRadioButton" name="rbtnDec">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>70</y>
             <width>96</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>十进制</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QRadioButton" name="rbtnHex">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>100</y>
             <width>96</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>十六进制</string>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="grpRegister">
       <property name="title">
        <string>寄存器</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
